app/js/views/microphone.jsx

```diff
@@ -27,17 +27,56 @@ export default class Microphone extends React.Component {
     this.click = this.click.bind(this);
   }
 
+  loadAudio() {
+    const AudioContext = window.AudioContext || window.webkitAudioContext;
+    this.audioCtx = new AudioContext();
+
+    fetch('media/cue.wav')
+      .then((res) => {
+        if (!res.ok) {
+          throw new Error(res.status);
+        }
+
+        return res.arrayBuffer();
+      })
+      .then((arrayBuffer) => {
+        this.audioCtx.decodeAudioData(arrayBuffer, (buffer) => {
+          this.audioBuffer = buffer;
+        }, (err) => {
+          console.error('The audio buffer could not be decoded.', err);
+        });
+      });
+  }
+
+  playBleep() {
+    if (!this.audioCtx) {
+      return;
+    }
+
+    this.bufferSource = this.audioCtx.createBufferSource();
+    this.bufferSource.buffer = this.audioBuffer;
```

Comment:
Maybe you should also check `buffer` has been loaded and decoded before you create the source node. Some implementations might be picky and throw an error if you send them a null instead of a proper buffer.